walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,18 +62,46 @@
     }
     
     List<Event> mandatoryAttendeeEvents = new ArrayList<Event>();
+    List<Event> optionalAttendeeEvents = new ArrayList<Event>();
     for (Event event : allEvents) {
       if (!optionalAttendeesOnly(event, mandatoryRequestAttendees)) {
         mandatoryAttendeeEvents.add(event);
+      } else {
+        optionalAttendeeEvents.add(event);
       }
     }
     sortByStart(mandatoryAttendeeEvents);
+    sortByStart(optionalAttendeeEvents);
     
-    if (!mandatoryAttendeeEvents.isEmpty()) {
-      return getAllPossibleTimes(mandatoryAttendeeEvents, request);
+    if (mandatoryAttendeeEvents.isEmpty()) {
+      return everyoneTimeOptions;
+    }
+    
+    ArrayList<TimeRange> mandatoryAttendeeTimeOptions = getAllPossibleTimes(mandatoryAttendeeEvents, request);
+    
+    List<ArrayList<TimeRange>> possibleTimeOptionsCollection = new ArrayList<ArrayList<TimeRange>>();
```

Comment:
You can drop the Collection suffix here.

Reply:
Done